### ISSUE TITLE
documentation for TF models mentions non-existent methods

### ISSUE THREAD

**Issue by btel** [huggingface/transformers#2906]
Documentation of `TFPreTrainedModel.from_pretrained` method mentions the `.model()` and `.eval()` methods that are not defined for tensorflow models:

> The model is set in evaluation mode by default using ``model.eval()`` (Dropout modules are deactivated)
> To train the model, you should first set it back in training mode with ``model.train()``

https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_tf_utils.py#L195
